Automattic/publish-to-s3-gradle-plugin
# Address Android Studio warnings & improve README

There were quite a few warnings in Android Studio while I was working on https://github.com/Automattic/publish-to-s3-gradle-plugin/pull/33 and I wanted to take a moment to address them all before we release `0.9.0`.

[In this comment](https://github.com/Automattic/publish-to-s3-gradle-plugin/pull/33#discussion_r1296503582), I claimed that moving source files to a different folder would be harder to review alongside other changes - and yet, I've done that in this PR 😓 I still would have preferred to do this as a separate PR, but the release of `0.9.0` is a bit of a blocker for me, so I didn't want to delay it any longer.

I've made each type of change in a separate commit, so if the PR is reviewed commit by commit, it should be very easy to follow along. Most of the changes were suggested by Android Studio with the only extra change being the update of `README`. I felt that it got a bit outdated, so I tried to improve it by using the same syntax that we use in our libraries and adding a note about the `publish-to-s3-without-sources` plugin.

**To Test**

Due to the simplicity of the changes, I didn't go through any rigorous testing. However, I published `0.9.0-rc-26` version from the 997e29ee1b3586ad40d6e5dd33e37be3a3a9e07e commit and used that version to verify that the artifacts were successfully published in this PR: https://github.com/wordpress-mobile/WordPress-Utils-Android/pull/131
